xbmc/guilib/GUIControl.cpp

```diff
@@ -125,6 +125,18 @@ void CGUIControl::DoProcess(unsigned int currentTime, CDirtyRegionList &dirtyreg
   if (Animate(currentTime))
     MarkDirtyRegion();
 
+  // if the control changed culling state from true tto false, mark it
+  const bool culled = m_transform.alpha <= 0.01f;
+  if (m_isCulled && !culled)
```

Comment:
can be simplified:

```suggestion
  if (m_isCulled && !culled)
  {
    MarkDirtyRegion();
  }
  m_isCulled = culled;
```

Reply:
No, it can't be simplified. If m_isCulled is true, MarkDirtyRegion() bails.

Should I add a new function MarkDirtyRegionIgnoreCulling() to make it clearer?

Also, it should mark both transitions as dirty (e.g. `m_isCulled != culled`).

Reply:
I mean you don't need the if/else check as you'll be updating m_isCulled regardless of it being false or true, afaics. Snippet I posted is exactly identical to yours.

Reply:
Nvm, I see what you mean now